### ISSUE TITLE
Deprecate EntityManager::clear with argument of one entity class and EntityRepository::clear 

### ISSUE THREAD

**Issue by beberlei** [doctrine/orm#8460]
It is possible to call `EntityManager::clear($entity)` or `EntityRepository::clear()` to only clear one kind of entity from the identity map. this is done by calling detach on the entity instead of actually clearing the identity map and other related `UnitOfWork` data.

Because clearing is not semantically the same as detaching an entity, it is not possible to do this anymore.

Alternative: If you need to detach many entities, you can perform this operation directly using the UnitOfWork. Potentially hide this away in a service or repository:

```php
$unitOfWork = $entityManager->getUnitOfWork();
$entities = $unitOfWork->getIdentityMap()[Entity::class] ?? [];

foreach ($entities as $entity) {
    $entityManager->detach($entity);
}
```

**Comment by gempir** [comment]
The documentation for batch processing will need an update then 

https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/batch-processing.html#iterating-results

**Comment by derrabus** [comment]
Fixed via #9471.
